### ISSUE TITLE
RC2 - Release Notes + Risk Sections

### ISSUE THREAD

**Issue by ossie-git** [OWASP/Top10#131]
https://github.com/OWASP/Top10/blob/master/2017/en/0x06-ReleaseNotes.md 

"Although most of these issues are weaknesses, the last one is missing or ineffective controls"

"the last one is missing or ineffective controls".  Sounds strange.  Perhaps "the last one is a missing / ineffective control"

"With firewalls, logging, network IPS and endpoint protection systems unable to detect web hacks, it falls on web apps themselves to be the new firewall, especially for cloud-based microservices, where there is no firewall, no IPS, and no end points."

Ouch.  This looks like a workaround and out of place.  Anyone would logically say, yeah, but my WAF or RASP solution will protect me and they would be partially right (they would protect them a lot more than network-security devices).  Mentioning network and endpoint protection systems is a little out of context + even then, they can provide some level of protection and detection (look at NSM / Bor and what you can do with netflows, etc.)

I would suggest rewording the above sentence.

I think you should also remove the word logging after firewall as it is in the title of the point

Also, as the focus is on logging, things like WAFs and even RASPs can give you 100% visibility (unless you're doing something with websockets and some custom protocol it doesn't support)

So I think you are beating around the bush by not mentioned them.

I would suggest switching the order of "Risk Factors" and "About Risks" as it makes more sense and makes understanding the "Risk Factors" easier

Also, not sure why you removed the image from - https://www.owasp.org/index.php/Top_10_2013-Risk

**Comment by jmanico** [comment]
I think the goal of this T10 item is to encourage developers to do purposeful security logging in applications. No product can replace intentional logging in programming by developers.

I do think we should also encourage developers to work with incident response teams so everyone is in sync about •what• is getting logged.

Consider referencing https://www.owasp.org/index.php/Category:OWASP_Logging_Project as well, which will help developers flag log entries as security centric log entries (and more).

Aloha, Jim

**Comment by ossie-git** [comment]
I agree that nothing will substitute for purposeful security logging in a format that is easily digestable by SOC teams to provide the necessary coverage.  But this area is currently very weak (AppSensor is basically the only major project in this area and it is kind of bare bones)

While ideally this is the right approach, having additional controls that can give you at least the needed visibility for IR work is a definite plus.

But mentioning network level controls in this section when you have application level controls that I agree are not ideal (but are a lot better than the network ones for logging + protecting web applications) doesn't make a lot of sense.

I think it can be reworded to place the onus on developers while also involving others in secondary layers related to T10

Realistically, using additional controls can help immensely and shouldn't be neglected, especially as this is a new entry.  Hopefully in 3 years time when the standard is updated, frameworks, libraries, and developers would we able to do this primarily without any secondary controls.

Anyway, off to sleep.  Will complete the document tomorrow and send my feedback for the rest of the issues.

**Comment by jmanico** [comment]
Better link for this one is https://www.owasp.org/index.php/OWASP_Security_Logging_Project

--
Jim Manico
@Manicode
Secure Coding Education
+1 (808) 652-3805

**Comment by vanderaj** [comment]
We will have to address this issue post RC2 as we're really close to freezing RC2 text.

**Comment by vanderaj** [comment]
@ossie-git , can you mail me vanderaj at owasp.org with your real name if you want to be included in the Contributors page. Your QA's on such short notice are frankly amazing, and I want to include all of the improvements where I can, and you should totally get kudos for it.

**Comment by jmanico** [comment]
+1 Very surgical and helpful comments. Rock on!

--
Jim Manico
@Manicode

**Comment by ossie-git** [comment]
Thanks Andrew.

Sure.  My name is Osama Elnaggar.  Don’t know if you remember me but I sent
you my ASVS feedback earlier this year.

I’d appreciate it if you updated the acknowledgements for the Final release.

PS. I sent you an email directly to your email address but it looks like
you didn’t receive it.

-- 
Osama Elnaggar

On October 18, 2017 at 6:43:18 AM, Andrew van der Stock (
notifications@github.com) wrote:

@ossie-git <https://github.com/ossie-git> , can you mail me vanderaj at
owasp.org with your real name if you want to be included in the
Contributors page. Your QA's on such short notice are frankly amazing, and
I want to include all of the improvements where I can, and you should
totally get kudos for it.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/OWASP/Top10/issues/131#issuecomment-337347648>, or mute
the thread

**Comment by vanderaj** [comment]
@ossie-git The current release notes no longer have any of the dodgy text, so I'm closing this issue. When you are ready for another big review, let me know and I will generate a PPTX / markdown / PDF / Word version for you (whatever is easiest for you to consume).
